cmake/fuzzing_options.cmake

```diff
@@ -0,0 +1,44 @@
+# Copyright 2018 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Adds fuzzing options to the current build.
+option(FUZZING "Build with Fuzz Testing flags" OFF)
+
+# If fuzzing is enabled, multiple compile and linking flags must be set.
+# These flags are set according to the compiler kind.
+if(FUZZING)
+  # Fuzzing must be accompanied by WITH_ASAN=ON.
+  if(NOT WITH_ASAN)
```

Comment:
Question: `WITH_ASAN` is required, but the coverage flag is set here. Can you add a comment explaining why these flags are treated differently?

Reply:
I wanted to make sure that ASAN is enabled if we are fuzzing. I modified the comment and the error message. Let me know if it makes sense.

Reply:
What I'm trying to get at is, why can't you do
```
if(NOT WITH_ASAN)
  set(WITH_ASAN true) # Pseudo-syntax, but you get the idea
endif()
```
Just add a short rationale on this, if possible.